README.md
# Crystalis

Crystalis is an extension for the [Enterprise Content Management Framework TYPO3 CMS](http://typo3.org/), released under
the GNU GPL. Crystalis is copyright (c) 2012–2016 by Daniel Haring.

This document covers installation and basic usage of Crystalis. A more in-depth documentation about available features
and how Crystalis works exactly can be found at:

* https://docs.typo3.org/typo3cms/extensions/crystalis/

## Preparation

Crystalis requires a working
[installation of TYPO3 CMS](https://docs.typo3.org/typo3cms/GettingStartedTutorial/Installation/Index.html), version 6.2
or higher.

Please make sure to have one up and running before continue reading.

It is assumed that you have at least basic knowledge about TYPO3 CMS. If you need support in installing and using TYPO3
CMS, please refer to the [Official Documentation](http://typo3.org/documentation/).

## What does it do?

Crystalis is intended to be used as a headstone for newly built websites using TYPO3 CMS. It is shipped with an own
content rendering logic, provides a HTML5 video content element, supports you in handling languages and makes some
smaller improvements which should lower the effort building your site.

## Available versions

For better transparency, the version number of Crystalis also indicates the TYPO3 CMS version required to run properly.

There will be versions for every TYPO3 CMS LTS ("Long term support") release which is currently supported at the
official level, and in addition one version for the most recent release of TYPO3 CMS.

The master branch represents the cutting-edge state of the extension and does not necessarily run flawlessly and in many
cases even is not intended to do so. Whereas the release branches are fully tested with the respective version of TYPO3
CMS and should work without downsides.

## Installation

Basically there are three ways to implement Crystalis into your TYPO3 CMS environment:

 - By using the official way through the TYPO3 Extension Repository (TER)
 - By downloading a ZIP-file out of the available branches/releases
 - By using git via command line

Please choose the method you're most comfortable with and continue reading.

### Installing via TYPO3 Extension Repository (TER)

The most intuitive way of installing Crysalis is by using the Extension Manager module of your TYPO3 CMS environment.

Please keep in mind, that there could only exist one recent version of Crystalis in TER. If you are using an older
release of TYPO3 CMS (e.g. 6.2 LTS) you should consider using one of the other methods.

Installing Crystalis via TER works as follows:

1. Log into the Backend of your TYPO3 CMS installation
2. Switch to module *Extension Manager*
3. From the select-box at the upper right corner, choose "Get Extensions"
4. In the search field at the top type "crystalis"
5. Click the "Import and Install" action left beside the extension name
6. TYPO3 CMS now will take care about installing and activating Crystalis

### Installing via ZIP-file

Installing Crystalis from a downloaded ZIP file has the advantage that you have access to the most recent release of
every branch/version.

To install Crystalis via ZIP-file please do the following:

1. Download the release branch of your choice (e.g. Crystalis-6.2)
2. Unpack the ZIP-file into the following directory of your TYPO3 CMS installation: *typo3conf/ext*
3. Rename the extracted folder to *crystalis* and ensure your web server has at least reading access to it
4. Log into the Backend of your TYPO3 CMS installation
5. Switch to module *Extension Manager*
6. Crystalis should appear at the lower half of the extension list. Click the "Activate" button on it's left and you're
   done.

### Installing via command line

Using the command line allows you to fine-tune access rights and even keep track of future changes made to Crystalis.

This method requires git to be installed.

To install Crystalis via command line on Unix like systems, please follow these steps:

Change to the extension directory of your TYPO3 CMS installation:
```bash
cd typo3conf/ext
```

Clone the desired branch of Crystalis (here: *Crystalis-6.2*):
```bash
git clone -b Crystalis-6.2 https://github.com/DanielHaring/TYPO3-Crystalis.git crystalis
```

(Optional) Remove git related files to avoid tracking:
```bash
rm -R crystalis/.git/
```

Grant write access to your web server (assuming the group name of your web server is '_www'):
```bash
chown -R :_www crystalis/ && chmod -R g+w crystalis/
```

Follow steps 4–6 of *Installing via ZIP-file*

ChangeLog
2016-03-23                           [RELEASE] Release of Crystalis 7.6.1 (Daniel Haring)
2016-03-23  a1c3e2d                  [BUGFIX] Prepare Database Connection for RealURL (Daniel Haring)
2016-03-23  003fcf0                  [BUGFIX] Cache generated Database Connection (Daniel Haring)
2016-03-22  4596b97                  [TASK] Update copyright comments (Daniel Haring)
2016-03-22  44ad5de                  [TASK] Update ReadMe file (Daniel Haring)
2016-03-22  5ba3b4b                  [TASK] Update documentation (Daniel Haring)
2016-03-21  602cd2d                  [TASK] Upgrade to RealURL v2.0.x (Daniel Haring)
2016-03-20  528c544                  [BUGFIX] Remove video dimension limits (Daniel Haring)
2016-03-20  b11c301                  [BUGFIX] Update Flash Player urls (Daniel Haring)
2016-03-20  0bf9e68                  [BUGFIX] Use getter for retrieving Resource Factory (Daniel Haring)
2016-03-20  9bd3081                  [TASK] Add Unit Tests for DatabaseService (Daniel Haring)
2016-03-20  964554e                  [TASK] Add Unit Tests for RealurlConfigurator (Daniel Haring)
2016-03-20  8ab0301                  [TASK] Allow control of the RealURL Auto-Configurator (Daniel Haring)
2016-03-20  778b708                  [TASK] Add possibility to disable cache (Daniel Haring)
2016-03-20  acb3ad8                  [BUGFIX] Remove Object Manager dependency (Daniel Haring)
2016-03-20  19b054e                  [BUGFIX] Inject Rewrite Configurators via makeInstance (Daniel Haring)
2016-03-20  5b542f2                  [BUGFIX] Use own IsoCodeService implementation (Daniel Haring)
2016-03-20  d3d44a2                  [TASK] Add Unit Tests for ConfigurationUtility (Daniel Haring)
2016-03-20  09e3f76                  [TASK] Add Unit Tests for Language Service (Daniel Haring)
2016-03-20  c2cc600                  [TASK] Allow access to used Database Services (Daniel Haring)
2016-03-19  c66a9b5                  [TASK] Outsource Rewrite Configurator management (Daniel Haring)
2016-03-19  65ae6a6                  [BUGFIX] Initialize Language Service if not present (Daniel Haring)
2016-03-19  8977488                  [BUGFIX] Avoid namespace conflicts (Daniel Haring)
2016-03-19  9e4c307                  [TASK] Implement basic Unit Tests (Daniel Haring)
2016-03-17  93d98de                  [TASK] Compare instances using instanceof (Daniel Haring)
2016-03-17  295ef64                  [TASK] Prefer fully qualified class names in comments (Daniel Haring)
2016-03-17  c7bf820                  [CLEANUP] Remove obsolete database check (Daniel Haring)
2016-03-17  2fe7f7c                  [TASK] Correct typos (Daniel Haring)
2016-03-13  09684bc                  [TASK] Correct Readme typos (Daniel Haring)
2016-03-13  d1a1679                  [TASK] Specify FLUID schemas (Daniel Haring)
2016-03-13  86a9bdc                  [BUGFIX] Remove 'ToDo' from documentation index (Daniel Haring)
2016-03-13  ed02160                  [!!!][TASK] Set vendor name to DanielHaring (Daniel Haring)
2016-03-12  11f932a                  [TASK] Update GitHub urls (Daniel Haring)
2016-03-12  a1e51c1                  [BUGFIX] Use Core Database Connection (Daniel Haring)
2016-03-12  0d88ed7                  [BUGFIX] Specify Language Service namespace (Daniel Haring)
2016-03-12  02f1ad0                  Revert "[CLEANUP] Remove obsolete closing tags" (Daniel Haring)
2016-03-12  aa646f1                  [TASK] Shorten unecessary FQCNs (Daniel Haring)
2016-03-12  512aa85                  [BUGFIX] Define configurator check as non-static (Daniel Haring)
2016-03-12  af1009b                  [CLEANUP] Remove obsolete closing tags (Daniel Haring)
2016-03-12  6a412a6                  [BUGFIX] Call Resource Factory by getter (Daniel Haring)
2016-03-12  2151a10                  [TASK] Implement the Resource Factory (Daniel Haring)
2016-03-12  0ceb329                  [TASK] Restrict supported RealUrl version (Daniel Haring)
2016-03-12  1155f0d                  [CLEANUP] Remove obsolete use statements (Daniel Haring)
2016-03-03  fdb29f1                  [BUGFIX] Apply image setup as copy (Daniel Haring)
2016-03-03  6fcd359                  [TASK] Specify composer dependencies (Daniel Haring)
2016-03-03  c83dc9f                  [TASK] Update License (Daniel Haring)
2016-02-17  f28b53a                  [TASK] Set Crystalis version to 7.6.1-dev (Daniel Haring)

2016-01-23                           [RELEASE] Release of Crystalis 7.6.0 (Daniel Haring)
2016-01-23  f436af6                  [TASK] Update Documentation (Daniel Haring)
2016-01-23  5106cda                  [TASK] Update Readme (Daniel Haring)
2016-01-23  1db9a73                  [TASK] Allow ExtBase to alter meta tags (Daniel Haring)
2015-12-08  bd059f3                  [TASK] Make use of ::class when referencing class names (Daniel Haring)
2015-12-08  906e007                  [CLEANUP] Remove deprecated calls to SpriteManager (Daniel Haring)
2015-12-08  5da8179                  [CLEANUP] Remove obsolete typeicons (Daniel Haring)
2015-12-08  f5f9a3a                  [TASK] Update TCA link wizard name (Daniel Haring)
2015-12-08  8d672cc                  [TASK] Define render mode for TCA fields of type 'select' (Daniel Haring)
2015-12-08  095cf9f                  [TASK] Update icon names for TCA items (Daniel Haring)
2015-12-08  de9a704                  [TASK] Update TCA icon paths (Daniel Haring)
2015-12-08  89ca00a                  [TASK] Upgrade to TYPO3 CMS v7.6.x (Daniel Haring)
2015-12-08  a3fcb2d                  [TASK] Set Crystalis version to 7.6.0-dev (Daniel Haring)
2015-10-18  d247770                  [TASK] Render spaces as empty paragraph (Daniel Haring)
2015-10-18  59b601d                  [BUGFIX] Define additional RTE classes within textstyle (Daniel Haring)
2015-10-18  c2ac200                  [TASK] Move locale mapping to IsoCodeServlce (Daniel Haring)
2015-10-18  d5e8e94                  [BUGFIX] Force array for retrieveDomainInformations (Daniel Haring)

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "crystalis".
 *
 * Auto generated 23-03-2016 19:37
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
	'title' => 'Crystalis',
        'description' => 'A flexible headstone for professional next generation TYPO3 CMS websites. Crystalis provides content rendering, pre-configured setups, fully automatic language handling and many more. ',
	'category' => 'fe',
	'version' => '8.0.0-dev',
	'state' => 'stable',
	'uploadfolder' => 0,
	'createDirs' => '',
	'clearCacheOnLoad' => 1,
	'author' => 'Daniel Haring',
	'author_email' => 'development@haring.co.at',
	'author_company' => '',
	'constraints' =>
	array (
		'depends' =>
		array (
			'extbase' => '7.6.0-8.0.99',
			'fluid' => '7.6.0-8.0.99',
			'typo3' => '7.6.0-8.0.99',
		),
		'conflicts' =>
		array (
		),
		'suggests' =>
		array (
            'realurl' => '1.13.4-2.0.99',
            'felogin' => '7.6.0-8.0.99'
		),
	),
);

?>
